docs/contributing/documentation_guidelines.md

```diff
@@ -431,3 +431,27 @@ If you want to use a specific Sphinx directive, consult with core contributors t
 ### Arbitrary HTML
 
 While our documentation tooling offers some support for embedding arbitrary HTML, this is frowned upon. Only do so if there is a necessity, and if the formatting is unlikely to need updates.
+
+### Code example considerations
+When including code examples, particularly JavaScript or embedded HTML, it's important to follow best practices for security and accessibility. Here are some key considerations:
+
+1. Content Security Policy (CSP) Compliance
+When adding JavaScript from external sources or inline scripts, ensure CSP compliance to prevent security vulnerabilities like cross-site scripting (XSS). Avoid mark_safe where possible, and use format_html or external files to manage scripts securely. Additionally, verify that external resources are served over HTTPS and consider loading them from trusted sources or your own server where feasible.
+
+2. Accessibility Compliance
+Make sure that all examples are accessible and adhere to accessibility standards (e.g., WCAG). For interactive components, ensure proper keyboard navigation and screen reader support. When creating dynamic content or effects (such as animations or notifications), provide options for users to pause, stop, or adjust these features as needed.
+
+<!-- Code Quality and Security: Highlight the importance of following security guidelines, like sanitizing any user-generated content before rendering it in code examples.
+Encourage the use of ESLint or similar tools for JavaScript to help catch errors or accessibility issues early in the development process. But this all might be extra or too much.
+
+Clarity in Documentation Examples: Recommend using simple, clear examples that prioritize readability, and consider providing comments within code blocks where appropriate to clarify complex sections. However I think the docs do a great job of this already but theres not many comments in functions in my opinion. Ensure that examples are self-contained and avoid dependencies that may be difficult for newcomers to replicate or understand. 
+
+Performance and Compatibility: Maybe suggest using efficient code, testing in multiple browsers, and providing polyfills if needed for older browser support. I'm not sure how much complex we want the docs to be
+
+Error Handling and Versioning: Maybe include basic error handling in interactive examples to protect the copy-pasters, and specify dependency versions to ensure stability and reproducibility unless I missed them.
+
+-->
+
```

Comment:
I'll leave these out for now but great suggestions, maybe something to consider for the future.

We may even want to say something like 'remember the AI is watching :D and example code will become the code of the future' haha.